### ISSUE TITLE
Explicit Uptime / HTTPS / Obs values in advanced mode

### ISSUE THREAD

**Issue by slasherfun** [TheKinrar/instances#117]
Hi,

The advanced mode shows values for uptime / HTTPS / Obs, but with no explaination of what they are:
- uptime: over which period is shown the value?
- HTTPS: what does the letter mean? (the "legacy" page shows it)
- Obs: what does it mean / what does the letter mean? (the "legacy" page shows both)

**Comment by nedbat** [comment]
Is there a place in the code, or in the Mastodon/ActivityPub protocols that we can find the definitions?

**Comment by slasherfun** [comment]
HTTPS score is from https://cryptcheck.fr/
Obs (Observatory by Mozilla) score is from https://observatory.mozilla.org/analyze/

**Comment by jakawell** [comment]
Seems a lot of servers get Fs and Es for HTTPS, even info sec focused ones. Does SSL Labs have a free API? Because in my experiences, that is a much more widely used standard.

**Comment by doncruse** [comment]
I encountered this problem today with my instance. Cryptcheck.fr scores it an "F," but the specific problem that it cites seems to be a phantom. Other sites score it as an "A." It would be great if your project used a different source.

Unfortunately, in answer to @jakawell , it looks like the SSL Labs API terms might not permit this use (without some special variance being granted by SSL Labs): "You are allowed to: (i) use the API only to inspect only sites and servers whose owners have given you permission to do so;..." (from https://www.ssllabs.com/downloads/Qualys_SSL_Labs_Terms_of_Use.pdf )

**Comment by TheKinrar** [comment]
Is is that hard to get something better than F, though? (Honest question. I know cryptcheck increased its difficulty recently but I don't really know how much.)
I get an A+ on mastodon.xyz and I didn't do anything really special. I believe I use [this](https://ssl-config.mozilla.org) config with "Intermediate" settings.

**Comment by TheKinrar** [comment]
As for the *info sec focused* instances, infosec.exchange, which is the biggest one I think, gets an A+.

**Comment by doncruse** [comment]
You're right. They have made some tweaks to the default deployment setup. If I were running my own instance on my own hardware, I might tweak my nginx.conf as well (to "solve" this problem).

My concern is that using cryptcheck means the default nginx.conf that comes with mastodon/mastodon scores an "F." That means that any server host — like the one I use — who does diligently keep current with the main project is penalized for keeping current by scoring an "F."

Consider it this way: The letter-grade score is completely useless to a visitor of "instances.social" if it doesn't distinguish between a well-run and conscientious host that _does_ keep up-to-date with the main mastodon project and some other host that just threw together a deployment without worrying about security at all. As things stand, "instances.social" can't distinguish between those two situations at all. It gives an "F" to the main mastodon project. So even if it also gives an "F" to sketchy projects with bad practices, there's no way for a casual visitor to tell the difference between any of these "F" grades. They all score "F."

If you're curious, this is the nginx.conf that gets scored "A" on some others but "F" by cryptcheck: https://github.com/mastodon/mastodon/blob/53817294fc95eabfed6129138f9aaa920e13c4b9/dist/nginx.conf

If the view of this project is that it should be easy (for some values of "easy") for mastodon/mastodon to tighten up its own nginx.conf... I don't know if that's actually easy, across all the deployments of the main project. But that seems like something that should be addressed by communicating to that project, rather than scoring it and everyone else who deploys it an "F."

**Comment by TheKinrar** [comment]
To be clear, instances.social does not directly penalize instances based off their HTTPS score (anymore). I'm currently working on a rewrite of the wizard and lists, and I'll make it really clear this is more of a technical detail. Maybe I'll put more focus towards the Mozilla Observatory score as this one indirectly relies on SSL Labs (I think) and seems more consensual.

We could also revert back to using the old cryptcheck version (tls.imirhil.fr), or maybe I could parse the results and modify slightly the grading system (or modify it directly, I believe it's open source).

> But that seems like something that should be addressed by communicating to that project, rather than scoring it and everyone else who deploys it an "F."
Well, back in 2017-2018, adding this HTTPS ranking is precisely what increased the quality of HTTPS configs for many Mastodon instances, and trying to change the default config resulted in [that](https://github.com/mastodon/documentation/pull/474) kind of result... But we could maybe also try again, assuming it makes sense (I'm no crypto expert so I don't really know if cryptcheck is actually right or wrong).

**Comment by bostjanlaba** [comment]
We've managed to get scores to A+/A+ but our uptime is "wrong". It says 50+% while were online 99% of time. How is uptime calculated and over what period of time?

Thanks.
